kernel/comps/logger/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Logger System
+//!
+//! Users need to specify one and only one logger feature for this crate to enable
+//! the corresponding logger, with [`default_logger::DefaultLogger`] enabled by default.
+#![no_std]
+#![deny(unsafe_code)]
+
+extern crate alloc;
+
+use component::{init_component, ComponentInitError};
+
+mod console;
+#[cfg(feature = "default_logger")]
+mod default_logger;
+
+pub use console::_print;
+
+#[cfg(feature = "default_logger")]
+fn init_logger() {
+    default_logger::init();
+}
+
+#[init_component]
+fn init() -> Result<(), ComponentInitError> {
+    init_logger();
```

Comment:
I believe this crate cannot be compiled without the `default_logger` feature because of the undefined `init_logger` method.